controllers/ocsclient_controller.go

```diff
@@ -0,0 +1,475 @@
+		}
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// updateConsumerCapacity makes an API call to the external storage provider cluster to update the capacity
+func (r *OcsClientReconciler) updateConsumerCapacity(instance *v1alpha1.OcsClient, externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
+	instance.Status.Phase = v1alpha1.OcsClientUpdating
+
+	response, err := externalClusterClient.UpdateCapacity(
+		context.Background(),
+		instance.Status.ConsumerID,
+		instance.Spec.RequestedCapacity.String())
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			r.logGrpcErrorAndReportEvent(instance, UpdateCapacity, err, s.Code())
+		}
+		return reconcile.Result{}, err
+	}
+
+	responseQuantity, err := resource.ParseQuantity(response.GrantedCapacity)
+	if err != nil {
+		r.Log.Error(err, "Failed to parse GrantedCapacity from UpdateCapacity response.", "GrantedCapacity", response.GrantedCapacity)
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.RequestedCapacity.Equal(responseQuantity) {
+		klog.Warningf("GrantedCapacity is not equal to the RequestedCapacity in the UpdateCapacity response.",
+			"GrantedCapacity", response.GrantedCapacity, "RequestedCapacity", instance.Spec.RequestedCapacity)
+	}
+
+	instance.Status.GrantedCapacity = responseQuantity
+	instance.Status.Phase = v1alpha1.OcsClientConnected
+
+	return reconcile.Result{}, nil
+}
+
+/*
+// getExternalConfigFromProvider makes an API call to the external storage provider cluster for json blob
+func (r *OcsClientReconciler) getExternalConfigFromProvider(
+	instance *v1alpha1.OcsClient, externalClusterClient *providerClient.OCSProviderClient) ([]ExternalResource, reconcile.Result, error) {
+
+	response, err := externalClusterClient.GetStorageConfig(context.Background(), instance.Status.ConsumerID)
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			r.logGrpcErrorAndReportEvent(instance, GetStorageConfig, err, s.Code())
+
+			// storage consumer is not ready yet, requeue after some time
+			if s.Code() == codes.Unavailable {
+				return nil, reconcile.Result{RequeueAfter: time.Second * 5}, nil
+			}
+		}
+
+		return nil, reconcile.Result{}, err
+	}
+
+	var externalResources []ExternalResource
+
+	for _, eResource := range response.ExternalResource {
+
+		data := map[string]string{}
+		err = json.Unmarshal(eResource.Data, &data)
+		if err != nil {
+			r.Log.Error(err, "Failed to Unmarshal response of GetStorageConfig", "Kind", eResource.Kind, "Name", eResource.Name, "Data", eResource.Data)
+			return nil, reconcile.Result{}, err
+		}
+
+		externalResources = append(externalResources, ExternalResource{
+			Kind: eResource.Kind,
+			Data: data,
+			Name: eResource.Name,
+		})
+	}
+
+	return externalResources, reconcile.Result{}, nil
+}
+*/
+func (r *OcsClientReconciler) logGrpcErrorAndReportEvent(instance *v1alpha1.OcsClient, grpcCallName string, err error, errCode codes.Code) {
```

Comment:
We are not reporting any event from this function. Do we plan to do it in the future?